google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpanTest.java

```diff
@@ -286,22 +289,23 @@ public void singleUse() {
     Map<String, Boolean> spans = failOnOverkillTraceComponent.getSpans();
     assertThat(spans).containsEntry("CloudSpanner.ReadOnlyTransaction", true);
     assertThat(spans).containsEntry("CloudSpannerOperation.BatchCreateSessions", true);
-    assertThat(spans).containsEntry("SessionPool.WaitForSession", true);
     assertThat(spans).containsEntry("CloudSpannerOperation.BatchCreateSessionsRequest", true);
     assertThat(spans).containsEntry("CloudSpannerOperation.ExecuteStreamingQuery", true);
 
     List<String> expectedAnnotations =
         ImmutableList.of(
-            "Requesting 25 sessions",
-            "Request for 25 sessions returned 25 sessions",
-            "Creating 25 sessions",
+            "Requesting 2 sessions",
+            "Request for 2 sessions returned 2 sessions",
+            "Creating 2 sessions",
```

Comment:
Open Question for my understanding, given this is an unit test and not an integration test, I am assuming that the session pool functionality would be mocked.

So how does it matter if there are 2 sessions or 25 sessions? Do we need to mock a few session pool methods?

Reply:
The end call will be mocked, but these annotations are getting added at the time batch create session starts. So if the request is for 2 sessions, then the annotation will be added accordingly. 
`String.format("Requesting %d sessions", sessionCount)`